src/poetry/core/vcs/__init__.py

```diff
@@ -28,6 +28,7 @@ def get_vcs(directory: Path) -> Git | None:
                 [executable(), "rev-parse", "--show-toplevel"],
                 stderr=subprocess.STDOUT,
                 text=True,
+                encoding="utf-8",
```

Comment:
why can we assume that git produces utf8 output?

Reply:
@dimbleby Thanks for the feedback, I updated the encoding to locale.

Reply:
Now, we get a warning in the tests. Shall we [suppress it](https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings) in this case?

Reply:
That seems to be a tricky one. On my Windows machine, this only works with "utf-8". Minimal example:

```
>>> subprocess.check_output('git rev-parse --show-toplevel')
b'C:/temp/r\xc3\xa9p\xc3\xb6\n'
>>> subprocess.check_output('git rev-parse --show-toplevel', text=True)
'C:/temp/rÃ©pÃ¶\n'
>>> subprocess.check_output('git rev-parse --show-toplevel', text=True, encoding='locale')
'C:/temp/rÃ©pÃ¶\n'
>>> subprocess.check_output('git rev-parse --show-toplevel', text=True, encoding='utf-8')
'C:/temp/répö\n'
```

But obviously in our CI it has to be "locale" (I added a test that succeeds on my machine but fails in our CI). I'm wondering if there is a universal solution?

Reply:
OK, the issue with the test was not about encoding but about running the test alone and running all tests. There was another test that left the git access in a broken state and thereby made the new test fail.

To put it in a nutshell: no difference between my machine and CI. It looks like we can just assume utf-8, here.